nautobot/extras/models/groups.py

```diff
@@ -431,3 +443,353 @@ def clean(self):
 
         # Validate `filter` dict
         self.clean_filter()
+
+    @classmethod
+    def generate_query_for_filter(cls, filter_field, value):
+        """
+        Return a `Q` object generated from the a `filter_field` and `value`.
+
+        :param filter_field:
+            Filter instance
+        :param value:
+            Value passed to the filter
+        """
+        q = models.Q()
+        if isinstance(filter_field, django_filters.MultipleChoiceFilter):
+            for v in value:
+                q |= models.Q(**filter_field.get_filter_predicate(v))
+        else:
+            q |= filter_field.get_filter_predicate(v)
+        return q
+
+    @classmethod
+    def generate_query_for_group(cls, group):
+        """
+        Return a `Q` object generated from all filters for a `group`.
+
+        :param group:
+            DynamicGroup instance
+        """
+        fs = group.filterset_class(group.filter, group.get_queryset())
+        q = models.Q()
+
+        for field_name, value in fs.data.items():
+            filter_field = fs.filters[field_name]
+            q &= cls.generate_query_for_filter(filter_field, value)
+
+        return q
```

Comment:
As a classmethod both patterns are possible. As an instancemethod, only the former is possible. Since this method needs to be called recursively (see `process_group_filters()`) it can't just be the former. Additionally, I found there was a need to be able to re-use the method without requiring an instance, so I converted it to a class method.